version/useragent_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024 Hemi Labs, Inc.
+// Use of this source code is governed by the MIT License,
+// which can be found in the LICENSE file.
+
+package version
+
+import "testing"
+
+func TestCreateUserAgent(t *testing.T) {
+	tests := []struct {
+		product  string
+		version  string
+		comments []string
+		want     string
+	}{
+		{
+			version:  "0.0.0",
+			comments: []string{"test", ""},
+			want:     "heminetwork/0.0.0 (test)",
+		},
+		{
+			product: "test",
+			version: "",
+			want:    "test",
+		},
+		{
+			product:  "test",
+			version:  "0.0.0",
+			comments: []string{"", ""},
+			want:     "test/0.0.0",
+		},
+		{
+			product:  "bfgd",
+			version:  "1.0.0",
+			comments: []string{"test"},
+			want:     "bfgd/1.0.0 (test)",
+		},
+		{
+			product:  "bssd",
+			version:  "1.0.0",
+			comments: []string{"", "test", ""},
+			want:     "bssd/1.0.0 (test)",
+		},
+		{
+			product:  "bssd",
+			version:  "1.0.0",
+			comments: []string{" ", "test", " "},
+			want:     "bssd/1.0.0 (test)",
+		},
+		{
+			product:  "popmd",
+			version:  "0.1.0",
+			comments: []string{"Hemi Labs", "linux/amd64", "+https://github.com/hemilabs/heminetwork"},
+			want:     "popmd/0.1.0 (Hemi Labs; linux/amd64; +https://github.com/hemilabs/heminetwork)",
+		},
+	}
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
```

Comment:
could we add a name here? like add a `name` field to the test config then
```go
t.Run(tt.name, ...)
```

Reply:
Yep, I can do that. Go does use #%02d for these by default, which lets you tell which test is which but this would probably be better.